anthropic-java-bedrock/src/main/kotlin/com/anthropic/bedrock/backends/BedrockBackend.kt

```diff
@@ -0,0 +1,591 @@
+        // Parse the "type" field to allow the construction of a server-sent
+        // event (SSE) that might look like this:
+        //
+        //   event: content_block_delta
+        //   data: {"type":"content_block_delta","index":0,"delta":{"type":"text_delta","text":"Hello"}}
+        //
+        // Print the SSE (with a blank line after) to the piped output stream to
+        // complete the translation process.
+        //
+        // A thread avoids deadlocking the pipe. If everything is on the same
+        // thread, a "read" that blocks waiting for more data to be written,
+        // would block the thread from executing the necessary "write" and cause
+        // a deadlock.
+        threadPool.execute {
+            responseInput.use { input ->
+                // "use" closes the piped output stream when done, which signals
+                // the end-of-file to the reader of the piped input stream.
+                pipedOutput.use { output ->
+                    // When fed enough data (see loop, below) to create a new
+                    // "Message", the "Consumer.accept" lambda here is fired.
+                    val messageDecoder = MessageDecoder { message ->
+                        val sseJson = String(
+                            Base64.getDecoder().decode(
+                                jsonMapper.readTree(message.payload)
+                                    .get("bytes").asText()
+                            )
+                        )
+                        val sseEventType = jsonMapper.readTree(sseJson)
+                            .get("type").asText()
+
+                        output.write(
+                            "event: $sseEventType\ndata: $sseJson\n\n"
+                                .toByteArray()
+                        )
+                        output.flush()
+                    }
+
+                    val buffer = ByteArray(4096)
+                    var bytesRead: Int
+                    while (input.read(buffer).also { bytesRead = it } != -1) {
+                        messageDecoder.feed(buffer, 0, bytesRead)
+                    }
+                }
+            }
+        }
+
+        return object : HttpResponse {
+            override fun statusCode(): Int = response.statusCode()
+
+            override fun headers(): Headers =
+                response.headers().toBuilder()
+                    .replace(HEADER_CONTENT_TYPE, CONTENT_TYPE_SSE_STREAM)
+                    .build()
+
+            override fun body(): InputStream = pipedInput
+
+            override fun close() = pipedInput.close()
+        }
+    }
+
+    /**
+     * Releases resources used by this backend when they are no longer needed.
+     * This method may block for up to 30 seconds in the unlikely event that
+     * some responses are still being prepared when this is called.
+     */
+    override fun close() {
+        threadPool.shutdown()
+        threadPool.awaitTermination(30, TimeUnit.SECONDS)
```

Comment:
Is this actually necessary? I think our `OkHttpClient` just calls `shutdown()` without `awaitTermination()`. Probably best to be consistent